akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -632,7 +634,7 @@ class ShardRegion(
 
   def initializeShard(id: ShardId, shard: ActorRef): Unit = {
     log.debug("Shard was initialized {}", id)
-    shards = shards.updated(id, shard)
+    startingShards -= id
```

Comment:
This set holds all the shards actors that have already been started, but not yet recovered.